sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/GroupService.java

```diff
@@ -13,24 +13,90 @@
 */
 package fr.insee.sugoi.core.service;
 
+import fr.insee.sugoi.core.exceptions.GroupAlreadyExistException;
+import fr.insee.sugoi.core.exceptions.GroupNotFoundException;
+import fr.insee.sugoi.core.exceptions.UserNotFoundException;
 import fr.insee.sugoi.core.model.PageResult;
 import fr.insee.sugoi.core.model.PageableResult;
 import fr.insee.sugoi.model.Group;
+import java.util.Optional;
 
 public interface GroupService {
 
+  /**
+   * Check if the group exist in the app (by name) and create it if it doesn't exist
+   *
+   * @param realm
+   * @param appName
+   * @param group
+   * @return the created group
+   * @throws GroupAlreadyExistException if the group already exist
```

Comment:
How do we know if the group exist (by name ? is it case sensitive?)

Reply:
for a group it's not case sensitive but for a user it depends of the implementation (file different from ldap)